Esg inicio page

ezee-sg opened the pull request:
Así es como ha quedado de momento la página de inicio. Hay algunos detalles que se pueden pulir, como la transición de las imágenes del carrusel. Otro aspecto que he visto es que a veces tardan demasido en cargar algunas playlists, igual habría que incorporar algo que indique que están cargandose.

ezee-sg committed:
Merge branch 'spotify-web-api-implementation' into esg-inicioPage

ezee-sg committed:
Esqueleto de la pagina de inicio

ezee-sg committed:
Cambiado carrusel para que sea responsive

- Falta arreglar el texto del carrusel para que se adapte bien a distintos anchos de pantalla.

ezee-sg committed:
Fix del texto del carrusel

ezee-sg committed:
Merge branch 'spotify-web-api-implementation' into esg-inicioPage

ezee-sg committed:
Añadida Pagina Inicio

ezee-sg committed:
Cambio pagina Inicio

ezee-sg committed:
Merge branch 'spotify-web-api-implementation' into esg-inicioPage

ezee-sg committed:
Cambiado el estilo de la pagina de inicio

ezee-sg committed:
Popular playlists en pagina de inicio

ezee-sg committed:
Añadidas nuevas playlists a la pagina de Inicio

ezee-sg committed:
Merge branch 'jbr-playList' into esg-inicioPage

ezee-sg committed:
Añadido onClick en los carruseles para navegar hacia las playlists

ezee-sg committed:
Merge branch 'main' into esg-inicioPage

ezee-sg committed:
Añadido archivo css para la pagina de inicio

ezee-sg committed:
Pequeño cambio estetico

adriantd3 reviewed (approved):
Yo creo que hay que probar con otro tipo de carousel. El actual, al ser tan alargado,  hace que todas las imágenes se vean pixeladas, resultando en algo un poco cutre.

El estilo que creo que puede quedar bien siendo las imágenes cuadradas es algo de este estilo (literalemente he buscado en google: image card carousel react)

https://www.youtube.com/watch?v=4tzMi6G7a7w

Apruebo porque en general la estructura esta bien pero creo que habría que reconsiderar lo de los carouseles

VctPerez commented on Dutify/src/components/locations/inicio.jsx:
```diff
@@ -0,0 +1,39 @@
+
+
+    const cargarPlaylists = async () =>{
+        setRecentPlaylists(await getUserPlaylists(token))
```
Lo de las listas recientes tenemos que hablar que hacer con eso, porque no se pueden obtener de la api. Y tambien le añadiria un limite de 6 o 9 cartas.

JoseBueno30 replied:
esto es verdad, en lo del limite estoy de acuerdo, y si no podemos conseguir las listas recientes igual hay que replantear el diseño un poco

VctPerez commented on Dutify/src/components/carousel/carousel.jsx:
```diff
@@ -1,47 +1,45 @@
+    <section className='carrusel-container'>
+      <div className='carrusel-box'>
+        <h5 className='carrusel-h5'>{name}</h5>
+        <div id={id} className="carousel" data-bs-ride="carousel">
```
Esto no tiene mucha importancia pero, imagino que será por bootstrap, la animación del carrusel se me hace un poco rara.

JoseBueno30 commented:
> Yo creo que hay que probar con otro tipo de carousel. El actual, al ser tan alargado, hace que todas las imágenes se vean pixeladas, resultando en algo un poco cutre.
> 
> El estilo que creo que puede quedar bien siendo las imágenes cuadradas es algo de este estilo (literalemente he buscado en google: image card carousel react)
> 
> https://www.youtube.com/watch?v=4tzMi6G7a7w
> 
> Apruebo porque en general la estructura esta bien pero creo que habría que reconsiderar lo de los carouseles

VctPerez commented on Dutify/src/components/carousel/carousel.jsx:
```diff
@@ -1,47 +1,45 @@
+              </div>
+            ))}
           </div>
+          <button className="carousel-control-prev carousel-button" type="button" data-bs-target={button_target} data-bs-slide="prev">
```
TAmbien le bajaría el tamaño a las flechas, las noto que ocupan demasiado.

VctPerez commented on Dutify/src/components/carousel/carousel.jsx:
```diff
@@ -1,47 +1,45 @@
+              <div key={playlist.id} className={`carousel-item ${index === 0 ? 'active' : ''}`}>
+                <img className="d-block img-fluid" src={playlist.imageUrl} alt={playlist.description} 
+                onClick={()=>{ClickHandler(playlist.id)}} />
+                <div className="carousel-caption d-md-block fw-bolder carrusel-caja-texto">
```
Otra cosa sin importancia, es que la caption no está del todo pegada abajo

JoseBueno30 commented:
La estructura general está bien pero las listas recientes deberían ser a lo sumo seis o así, de esa forma no quedan tantas listas en el inicio y no pierde sentido la página de listas.

Las flechas usaría unas que no tuvieran parte central (osea algo así < >)

Lo que dice @adriantd3 sobre los carruseles es verdad, se alargan mucho las imágenes, pero los que ha propuesto no me terminan de convencer. Habría que darle un par de vueltas porque es un problema que también afecta a las cards, que se cortan las imágenes.

Los títulos de las cosas los pondría con una fuente mayor y sin subrayar porque al estar subrayados casi parecen enlaces.

Además falta que el ratón cambie a puntero en los carruseles.

DavidBuenoCarmona commented:
Además de lo que han dicho los demas, la página de inicio debería verse la primera al hacer el login, en cambio lleva a una página vacía:
![ShareX_4YYAlASybS](https://github.com/JoseBueno30/Dutify/assets/124664414/cb277a1c-2c93-4109-a638-3d9b7a712664)

ezee-sg committed:
Algunas mejoras en la pagina de inicio

ezee-sg commented:
He arreglado algunas de las cosas que me habéis comentado, lo único que está pendiente de cambiar es el apartado de listas recientes y el carrusel como tal (si nos da tiempo más adelante).

VctPerez committed:
Cambio de listas de artistas a "This is ..."

VctPerez committed:
Cambio de etiqueta carrusel

ezee-sg committed:
Etiquetas más próximas a sus carruseles

- Ahora las etiquetas estás más próximas a sus carruseles con el objetivo de que no haya confusión entre ellas.
